logger.h

```diff
@@ -1,62 +1,123 @@
 #include <log4cplus/loggingmacros.h>
 #endif
 
-/** Crée une configuration par défaut pour le logger */
-inline void init_logger(){
+namespace navitia {
+
+// logger pattern example : [18-09-10 17:02:17,186] [INFO ] [NDC] - message file.cpp:248
+static const std::string console_pattern_ref = "[%D{%y-%m-%d %H:%M:%S,%q}] [%-5p] [%x] - %m %b:%L  %n";
+
+// syslog pattern example : Sep 11 14:16:11 hostname binary_name: [PID] [INFO] [NDC] - message file.cpp:261
+static const std::string syslog_pattern_ref  = "[%i] [%p] [%x] - %m %b:%L  %n";
+
+/**
+ * @brief init simple navitia logger
+ */
+inline void init_logger()
+{
     log4cplus::BasicConfigurator config;
     config.configure();
+
+    auto properties = log4cplus::helpers::Properties();
+    properties.setProperty("log4cplus.rootLogger", "DEBUG, console");
+    properties.setProperty("log4cplus.appender.console", "log4cplus::ConsoleAppender");
+    properties.setProperty("log4cplus.appender.console.layout", "log4cplus::PatternLayout");
+    properties.setProperty("log4cplus.appender.console.layout.ConversionPattern", console_pattern_ref);
+
+    log4cplus::PropertyConfigurator configurator(properties);
+    configurator.configure();
 }
 
-/** configure the logger with a level and a pattern, the output will be on stdout */
-inline void init_logger(const std::string& level, const std::string& pattern){
+/**
+ * @brief init navitia logger
+ *
+ * @param name The application name
+ * @param level The level of the logger (INFO, DEBUG, etc..)
+ * @param active_local_syslog Active syslog redirection
+ * @param pattern Overload pattern logger
+ */
+inline void init_logger(const std::string& name,
+                        const std::string& level,
+                        const bool active_local_syslog = false,
+                        const std::string& pattern = "")
+{
     log4cplus::BasicConfigurator config;
     config.configure();
+
     auto properties = log4cplus::helpers::Properties();
-    properties.setProperty("log4cplus.rootLogger", (boost::format("%s, default") % level).str());
-    properties.setProperty("log4cplus.appender.default", "log4cplus::ConsoleAppender");
-    properties.setProperty("log4cplus.appender.default.layout", "log4cplus::PatternLayout");
-    properties.setProperty("log4cplus.appender.default.layout.ConversionPattern", pattern);
+
+    if (active_local_syslog) {
+        properties.setProperty("log4cplus.rootLogger", (boost::format("%s, console, syslog") % level).str());
+        properties.setProperty("log4cplus.appender.syslog", "log4cplus::SysLogAppender");
+        properties.setProperty("log4cplus.appender.syslog.ident", name);
+        properties.setProperty("log4cplus.appender.syslog.facility", "local7");
+        properties.setProperty("log4cplus.appender.syslog.layout", "log4cplus::PatternLayout");
+        if (pattern.empty()) {
+            properties.setProperty("log4cplus.appender.syslog.layout.ConversionPattern", syslog_pattern_ref);
```

Comment:
In this case we need to add something to identify the the coverage. As of now we do something like this: 
```
{coverage}: {pattern}
```

Reply:
:+1: